monkey/monkey_island/cc/services/config.py

```diff
@@ -425,3 +426,20 @@ def get_config_propagation_credentials():
             ),
             "exploit_ssh_keys": ConfigService.get_config_value(SSH_KEYS_PATH, should_decrypt=False),
         }
+
+    @staticmethod
+    def format_flat_config_for_agent(config: Dict):
+        ConfigService._remove_credentials_from_flat_config(config)
+
+    @staticmethod
+    def _remove_credentials_from_flat_config(config: Dict):
+        fields_to_remove = {
+            "exploit_lm_hash_list",
+            "exploit_ntlm_hash_list",
+            "exploit_password_list",
+            "exploit_ssh_keys",
+            "exploit_user_list",
+        }
```

Comment:
Probably would be better to not pull them from the database: https://docs.mongodb.com/manual/tutorial/project-fields-from-query-results/#return-all-but-the-excluded-fields

Not 100% sure we can do it with current pymongo

Reply:
Agreed, but some other reformatting needs to be done to support the agent work anyway. And whatever we do here will probably have to be redone when we refactor and add a DAL. Additionally, we need to support the old and new configuration formats for a while until we refactor our exploiters, so we'd need two different mongo queries to support both formats.